### PR TITLE
Catalog improvements

### PR DESCRIPTION
**Pull request by kandl**
- Display forms are contained in the attribute metadata objects, we can extract them from there, no need to load them
- Refresh wiremock recordings
- Make loading of the catalog groups optional - for now, default to true (for backward compatibility)

JIRA: RAIL-4459

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-backend-bear/src/backend/workspace/catalog/factory.ts [hunk 1]
```diff
@@ -270,24 +279,20 @@ export class BearWorkspaceCatalogFactory implements IWorkspaceCatalogFactory {
 
         const bearCatalogAttributes = bearCatalogItems.filter(GdcCatalog.isCatalogAttribute);
         const attributeUris = bearCatalogAttributes.map((attr) => attr.links.self);
-        const displayFormUris = flatMap(bearCatalogAttributes, (attr) => {
-            const geoPins = attr.links.geoPinDisplayForms ?? [];
```
Comment (no23reason) [inline libs/sdk-backend-bear/src/backend/workspace/catalog/factory.ts:274]: So the geo display forms are now handled in the `extractDisplayFormsFromBearAttributes` right?

Comment (kandl) [reply libs/sdk-backend-bear/src/backend/workspace/catalog/factory.ts:274]: Yes

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9654/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11536/ : SUCCESS in 21m 02s

File: libs/sdk-backend-spi/src/workspace/ldm/catalog.ts [hunk 2]
```diff
@@ -234,6 +240,15 @@ export interface IWorkspaceCatalogFactoryMethods<TFactory, TOptions> {
...
     excludeTags(tags: ObjRef[]): TFactory;
 
+    /**
+     * Setup whether catalog should fetch groups.
+     * Default: true
+     *
+     * @param loadGroups - should fetch groups
+     * @returns catalog factory
+     */
+    withGroups(loadGroups: boolean): TFactory;
```
Comment (no23reason) [inline libs/sdk-backend-spi/src/workspace/ldm/catalog.ts:250]: We should make sure the `groups()` call prints some error, or even throws if the factory was set up with groups not loading. WDYT?

Comment (kandl) [reply libs/sdk-backend-spi/src/workspace/ldm/catalog.ts:250]: It makes sense, but this is just a merge to hotfix branch, could we keep it for my PR to master?

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2950

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2733/ : SUCCESS in 2m 46s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3036/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1874/ : SUCCESS in 15m 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11542/ : SUCCESS in 23m 00s

**Pull request merged**